sympy/physics/mechanics/tests/test_joint.py

```diff
@@ -0,0 +1,232 @@
+from sympy import sin, cos, Matrix, sqrt, pi, expand_mul, Abs, acos, S
+from sympy.core.symbol import symbols
+from sympy.physics.mechanics import dynamicsymbols, Body, PinJoint
+from sympy.physics.mechanics.joint import Joint
+from sympy.physics.vector import Vector, ReferenceFrame
+from sympy.testing.pytest import raises
+
+t = dynamicsymbols._t
+
+
+def test_Joint():
+    parent = Body('parent')
+    child = Body('child')
+    raises(TypeError, lambda: Joint('J', parent, child))
+
+
+def test_pinjoint():
+    P = Body('P')
+    C = Body('C')
+    l, m = symbols('l m')
+    theta, omega = dynamicsymbols('theta_J, omega_J', real=True)
+
+    Pj = PinJoint('J', P, C )
+    assert Pj._name == 'J'
+    assert Pj.parent() == P
+    assert Pj.child() == C
+    assert Pj.coordinates() == [theta]
+    assert Pj.speeds() == [omega]
+    assert Pj.kdes() == [omega - theta.diff(t)]
+    assert Pj._parent_axis == P.frame.x
+    assert Pj._child_joint.pos_from(C.masscenter) == Vector(0)
+    assert Pj._parent_joint.pos_from(P.masscenter) == Vector(0)
+    assert Pj._parent_joint.pos_from(Pj._child_joint) == Vector(0)
```

Comment:
Why aren't all these public? Can't they be publicly accessible but no modifiable?

Reply:
- PinJoint.parent_joint_point
- PinJoint.child_joint_point

Seems useful to be able to access these points easily.

Reply:
I personally think they are good being private, user can check kdes, speeds, coordinates and other things but shouldn't be aware of all this

Reply:
If the user wants to connect things to these points and they haven't generated the points themselves, there there is no way to access them. What if I want to apply a force to this point, but it is hidden in the class?

Reply:
@moorepants  What all things shall I make public and immutable?